compiler+runtime/src/cpp/jank/read/parse.cpp

```diff
@@ -849,8 +849,42 @@ namespace jank::read::parse
       return object_source_info{ wrapped, start_token, str_end };
     }
 
+    if(sym->name == "cpp")
+    {
+      auto str_result(next());
+
+      if(str_result.is_err())
+      {
+        return str_result;
+      }
+      else if(str_result.expect_ok().is_none())
+      {
+        return error::parse_invalid_reader_symbolic_value(
+          "The string literal after this '#cpp' is missing.",
+          { start_token.start, latest_token.end });
+      }
+
+      auto const str_end(str_result.expect_ok().unwrap().end);
+
+      if(str_end.kind != lex::token_kind::string)
+      {
+        return error::parse_invalid_reader_symbolic_value(
+          "The form after '#cpp' must be a string literal.",
+          { start_token.start, latest_token.end });
+      }
+
+      auto const str(expect_object<obj::persistent_string>(str_result.expect_ok().unwrap().ptr));
+
+      auto const wrapped(
+        make_box<obj::persistent_list>(std::in_place,
+                                       make_box<obj::symbol>("cpp/value"),
+                                       make_box(util::format("\"{}\"", str->data)).erase()));
```

Comment:
I think we'll need `util::escape` for the string. Check out my test suggestions below.